flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1689,6 +1690,24 @@ public static int subtractMonths(long t0, long t1) {
         return x;
     }
 
+    public static String dateAdd(String startDate, int daysNumber) {
+        long startMillSecond;
+        try {
+            startMillSecond = FORMATTER_CACHE.get(DATE_FORMAT_STRING).parse(startDate).getTime();
+        } catch (ParseException e) {
+            try {
+                startMillSecond =
+                        FORMATTER_CACHE.get(TIMESTAMP_FORMAT_STRING).parse(startDate).getTime();
+            } catch (ParseException parseException) {
+                return null;
```

Comment:
These methods should fail, rather than returning null. You can then use `fallibleMethodGen` in the codegen